crates/wasi-preview1-component-adapter/src/lib.rs

```diff
@@ -94,6 +94,11 @@ impl<T, E> TrappingUnwrap<T> for Result<T, E> {
     }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn reset_adapter_state() {
+    State::init(get_state_ptr())
```

Comment:
Could this handle the case where `get_state_ptr` returns null as well?

Reply:
Ah, good catch!